### ISSUE TITLE
MSYS/Cygwin paths are not supported as terminal links

### ISSUE THREAD

**Issue by orgads** [microsoft/vscode#214615]
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: 1.90.0
- OS Version: Windows 10

Steps to Reproduce:

When I compile with MSYS on the terminal, paths in warnings/errors are shown in MSYS/Cygwin style: `[/cygdrive]/d/Projects/test/foo.cpp:123`. Ctrl-clicking it opens the Quick open edit with the path, then I have to edit it, remove the leading slash, add a colon after the drive letter and press Enter.

Created PR #214435 to handle that, but it was rejected because the change was in a central component (path.ts), and was too risky.

**Comment by AnrDaemon** [comment]
`/cygdrive/` is not a stable prefix. It could be anything, down to bare `/`.
If you want a stable prefix, there's `/proc/cygdrive/`, but it's nearly useless for the intended purpose of parsing terminal links. You would simply never encounter this form in the wild.

**Comment by orgads** [comment]
Yes, I know, but it's the default mount for cygwin, so most users just have it as it is. For MSYS it's `/`, and my plan is to support both `/` and `/cygdrive/`. It can be extended in the future to discover the mount and translate accordingly, but that's really niche.

**Comment by AnrDaemon** [comment]
"Most users" !== "usable default".
You have to, either, make it configurable (a can of worms in itself), or run each path through cygpath (super slow), or, being a good guest, produce `file:` links relative to the project root. VS Code will pick them up, no issue.

**Comment by AnrDaemon** [comment]
(Also, if I would be coding such a translator, I would not check for just `/cygdrive/`, but for `/<prefix/>[a..z]/`.)

**Comment by johan-boule** [comment]
> run each path through cygpath (super slow)

If you only have to resolve the path when clicked, there's no performance issue.
The difficulty seems to be to find out which cygpath program to run, if the user had multiple cygwin/msys/git-bash installations.

**Comment by AnrDaemon** [comment]
If the user has multiple installations, it's up to the user to provide a path to the correct executable.
